### PR TITLE
Add route for changing password of current user

### LINKED ISSUE #28: As a Customer, I want to change my password
* Change password (only authenticated)

```
PUT /change-password
```

Parameters:

* Current password
* New password
* Verification

### PR DESCRIPTION
**Pull request by nnatter**
Closes #28

This PR adds a `sylius_shop_api_change_password` route with the path `/me/change-password`.
The route allows to change the password of the user that is currently logged in.

The PR does not implement any new action, but rather uses the existing `changePasswordAction` of the `UserController` of Sylius. This is the same strategy that is already used for the `sylius_shop_api_password_reset` route.

### CODE REVIEW

File: tests/Controller/Customer/ChangePasswordApiTest.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                "first": "new-pass",
+                "second": "new-pass"
+            }
+        }
+JSON;
+
+        $this->client->request('PUT', '/shop-api/me/change-password', [], [], self::CONTENT_TYPE_HEADER, $changePasswordData);
+
+        $response = $this->client->getResponse();
+        $this->assertResponseCode($response, Response::HTTP_NO_CONTENT);
```
Comment (mamazu) [inline tests/Controller/Customer/ChangePasswordApiTest.php]: Maybe we should try to log in after changing the password to assert that it really works.

**Comment by lchrusciel** [comment]
Hey @nnatter,

thanks a lot for the contribution! 

@mamazu as we don't add any new code here, maybe we should add it before 1.0? On the one hand, the risk is rather low, as the feature existed already and it is tested. On the other hand, we have feature freeze before the stable release.

**Comment by mamazu** [comment]
Yeah, quite a basic feature but as you said feature freeze. I would wait for 1.1.

**Comment by mamazu** [comment]
Closes #9

**Comment by mamazu** [comment]
Closes #28

**Comment by mamazu** [comment]
Thanks, @nnatter! :1st_place_medal:

**Pull request merged**